engine/src/test/java/io/camunda/zeebe/engine/processing/usertask/AssignUserTaskTest.java

```diff
@@ -0,0 +1,190 @@
+  private static final String PROCESS_ID = "process";
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  private static BpmnModelInstance process() {
+    return process(b -> {});
+  }
+
+  private static BpmnModelInstance process(final Consumer<UserTaskBuilder> consumer) {
+    final var builder =
+        Bpmn.createExecutableProcess(PROCESS_ID).startEvent().userTask("task").zeebeUserTask();
+
+    consumer.accept(builder);
+
+    return builder.endEvent().done();
+  }
+
+  @Test
+  public void shouldEmitAssigningEventForUserTask() {
+    // given
+    ENGINE.deployment().withXmlResource(process()).deploy();
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+    final long userTaskKey =
+        RecordingExporter.userTaskRecords(UserTaskIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst()
+            .getKey();
+
+    // when
+    final Record<UserTaskRecordValue> assignedRecord =
+        ENGINE.userTask().withKey(userTaskKey).withAssignee("foo").assign();
+
+    // then
+    final UserTaskRecordValue recordValue = assignedRecord.getValue();
+
+    Assertions.assertThat(assignedRecord)
+        .hasRecordType(RecordType.EVENT)
+        .hasIntent(UserTaskIntent.ASSIGNING);
+
+    Assertions.assertThat(recordValue)
+        .hasUserTaskKey(userTaskKey)
+        .hasTenantId(TenantOwned.DEFAULT_TENANT_IDENTIFIER);
+  }
+
+  @Test
+  public void shouldRejectAssignIfUserTaskNotFound() {
+    // given
+    final int key = 123;
+
+    // when
+    final Record<UserTaskRecordValue> assignedRecord =
+        ENGINE.userTask().withKey(key).withAssignee("foo").expectRejection().assign();
+
+    // then
+    Assertions.assertThat(assignedRecord).hasRejectionType(RejectionType.NOT_FOUND);
+  }
+
+  @Test
+  public void shouldAssignUserTaskWithAssignee() {
+    // given
+    ENGINE.deployment().withXmlResource(process()).deploy();
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    final Record<UserTaskRecordValue> assignedRecord =
+        ENGINE.userTask().ofInstance(processInstanceKey).withAssignee("foo").assign();
+
+    // then
+    Assertions.assertThat(assignedRecord)
+        .hasRecordType(RecordType.EVENT)
+        .hasIntent(UserTaskIntent.ASSIGNING);
+    Assertions.assertThat(assignedRecord.getValue()).hasAssignee("foo");
+  }
+
+  @Test
+  public void shouldUnassignUserTaskWithNoAssignee() {
+    // given
+    ENGINE.deployment().withXmlResource(process(b -> b.zeebeAssignee("foo"))).deploy();
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    final Record<UserTaskRecordValue> assignedRecord =
+        ENGINE.userTask().ofInstance(processInstanceKey).assign();
```

Comment:
:upside_down_face:  The user task is un-assigned since we're not passing any value to the `withAssignee(String)` method, and by default, this is an empty String.

I wonder if we can make this more explicit in the test.

Reply:
Hmm, would you rather add something like `.withoutAssignee()` in there to make that visible? Could be a method that does nothing or sets the assignee to `null` actually.

Reply:
Done with https://github.com/camunda/zeebe/pull/15573/commits/8b92fb2e26334af6e38c2d96a5f4ef9fc0366529